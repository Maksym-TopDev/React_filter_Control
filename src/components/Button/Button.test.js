import React from 'react';
import ReactDOM from 'react-dom';
import TestRenderer from 'react-test-renderer';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Button from './Button';

configure({ adapter: new Adapter() });

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Button />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('structures', () => {
  const testRenderer = TestRenderer.create(
    <Button text="Test" />,
  );
  expect(testRenderer.toJSON()).toMatchSnapshot();
});

it('onClick', () => {
  const spyClick = jest.fn();
  const button = shallow((
    <Button onClick={spyClick} />
  ));
  button.find('button').simulate('click');
  expect(spyClick).toHaveBeenCalledTimes(1);
});
