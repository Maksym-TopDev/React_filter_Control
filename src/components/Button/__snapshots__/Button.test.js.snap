// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`structures 1`] = `
<button
  className="fc-button"
  type="button"
>
  Test
</button>
`;
